impl-base/src/main/java/org/jboss/shrinkwrap/impl/base/URLPackageScanner.java

```diff
@@ -140,10 +148,10 @@ private void handleArchiveByFile(File file) throws IOException, ClassNotFoundExc
             while (entries.hasMoreElements()) {
                 ZipEntry entry = entries.nextElement();
                 String name = entry.getName();
-                if (name.startsWith(packageNamePath) && name.endsWith(".class")
-                    && (addRecursively || !name.substring(packageNamePath.length() + 1).contains("/"))) {
-                    String className = name.replace("/", ".").substring(0, name.length() - ".class".length());
-                    foundClass(className);
+                if (name.startsWith(prefix + packageNamePath) && name.endsWith(".class")
+                        && (addRecursively || !name.substring((prefix + packageNamePath).length() + 1).contains("/"))) {
+                    String className = name.replace("/", ".").substring(prefix.length(), name.length() - ".class".length());
```

Comment:
No need @bobmcwhirter I will finish the PR tonight.  All code looks good, I
have made some minor commits of my own; I just need to wrap my head around
the problem and understand the reasoning behind your approach here.